src/Messaging/Messaging.test.ts

```diff
@@ -0,0 +1,142 @@
+import * as amqplib from 'amqplib'
+import { describe } from 'riteway'
+import { stub, spy } from 'sinon'
+import { Exchange } from '../Messaging/Messages'
+import { Messaging } from './Messaging'
+
+const createStubConnect = () => {
+  const assertExchange = spy()
+  const publish = spy()
+  const bindQueue = spy()
+  const consume = spy()
+  const assertQueue = stub().returns({ queue: 'queue' })
+  const createChannel = stub()
+  createChannel.returns({ assertExchange, publish, assertQueue, bindQueue, consume })
+
+  const customEvents = {
+    createChannel,
+    on: () => {},
+  }
+
+  return {
+    connect: () => Promise.resolve(customEvents),
+    createChannel,
+    assertExchange,
+    publish,
+    assertQueue,
+    bindQueue,
+    consume,
+  }
+}
+
+describe('Messaging', async (should: any) => {
+  const { assert } = should('')
+
+  {
+    const { connect, createChannel } = createStubConnect()
+    const stubConnect = stub(amqplib, 'connect').callsFake(connect)
+    const messaging = new Messaging()
+    await messaging.start()
+
+    assert({
+      given: 'start() with valid connectionUrl',
+      should: 'call createChannel()',
+      actual: createChannel.calledOnce,
+      expected: true,
+    })
+    stubConnect.restore()
+  }
+
+  {
+    const messaging = new Messaging()
+    const expected = 'Unable to connect to RMQ at amqp://localhost. connect ECONNREFUSED 127.0.0.1:5672'
+    assert({
+      given: 'start() with invalid connectionUrl',
+      should: `return the error: ${expected}`,
+      actual: await messaging.start().catch(e => e.message),
+      expected,
+    })
+  }
+
+  {
+    const { connect, assertExchange, publish } = createStubConnect()
```

Comment:
I'm sorry but I don't see how these unit tests add value. They look more complex than the code being tested — which is 99.9% amqplib and 0.01% our own code.

They also seem to focus too much on testing that the functions are called with this or that parameter (which is an implementation detail) rather than checking that they actually work.

Communication across modules should be tested with integration tests IMO. ([And it already is](https://github.com/poetapp/node/blob/master/test/Integration/PostWork.ts#L42))